components/brave_today/browser/brave_news_p3a_unittest.cc

```diff
@@ -233,20 +227,151 @@ TEST_F(BraveNewsP3ATest, TestTotalCardsViewed) {
 
   task_environment_.AdvanceClock(base::Days(4));
 
-  ResetCurrSessionTotalViewsCount(prefs);
+  RecordAtSessionStart(prefs);
   RecordTotalCardViews(prefs, 15);
   histogram_tester_.ExpectBucketCount(kTotalCardViewsHistogramName, 4, 1);
 
-  ResetCurrSessionTotalViewsCount(prefs);
+  RecordAtSessionStart(prefs);
   RecordTotalCardViews(prefs, 15);
   histogram_tester_.ExpectBucketCount(kTotalCardViewsHistogramName, 5, 1);
 
   task_environment_.AdvanceClock(base::Days(4));
 
-  ResetCurrSessionTotalViewsCount(prefs);
+  RecordAtSessionStart(prefs);
   RecordTotalCardViews(prefs, 0);
   histogram_tester_.ExpectBucketCount(kTotalCardViewsHistogramName, 4, 2);
 }
 
+TEST_F(BraveNewsP3ATest, TestLastUsageTime) {
+  PrefService* prefs = GetPrefs();
+  RecordAtInit(prefs);
+  // Should not report if News was never used
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 0);
+
+  RecordAtSessionStart(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 1);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 1, 1);
+
+  task_environment_.AdvanceClock(base::Days(7));
+  RecordAtInit(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 2);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 2, 1);
```

Comment:
I guess this means `RecordAtInit` runs at startup even if News is disabled?

Reply:
yes